Monika After Story/game/cardgames.rpy

```diff
@@ -0,0 +1,5639 @@
+                        and self.reactions[-1]["type"] == self.game.MONIKA_REFLECTED_ACT
+                    ):
+                        reaction = {
+                            "type": self.game.PLAYER_REFLECTED_ACT,
+                            "turn": self.game.current_turn,
+                            "monika_card": None,
+                            "player_card": self.game.game_log[-2]["played_card"],
+                            "shown": False
+                        }
+
+                        seen_count = self.reactions[-1]["seen_count"] + 1# use seen_count + 1 from the previous MONIKA_REFLECTED_ACT reaction
+                        reaction["seen_count"] = seen_count if seen_count < 3 else 2
+
+                        self.reactions.append(reaction)
+
+                        return reaction
+
+                    # Monika played an act > the player reflected > Monika failed to reflect
+                    elif (
+                        len(self.game.game_log) > 2# did we play enough turns?
+                        and self.game.game_log[-3]["played_card"] is not None# did Monika played a card back then?
+                        and self.game.game_log[-3]["played_card"].label == self.game.game_log[-2]["played_card"].label# Does it have the same label as the player's last cart?
+                    ):
+                        reaction = {
+                            "type": self.game.PLAYER_REFLECTED_ACT,
+                            "turn": self.game.current_turn,
+                            "monika_card": None,
+                            "player_card": self.game.game_log[-2]["played_card"],
+                            "seen_count": 0,# for this one always use seen_count 0
+                            "shown": False
+                        }
+
+                        self.reactions.append(reaction)
+
+                        return reaction
+
+                    # NOTE: There's a possibility that we didn't return yet,
+                    # so we'll have to fall through remaining checks
+
+                # # # Monika mirrors a WDF card
+                if (
+                    next_card_to_play is not None
+                    and next_card_to_play.label == "Draw Two"# did Monika played a d2 to reflect the wd4?
+                    and self.should_skip_turn# and she has to skip her turn
+                    and len(self.game.game_log) > 1
+                    and self.game.game_log[-2]["played_card"] is not None
+                    and self.game.game_log[-2]["played_card"].label == "Wild Draw Four"# and the player played a wd4
+                ):
+                    reaction = {
+                        "type": self.game.MONIKA_REFLECTED_WDF,
+                        "turn": self.game.current_turn,
+                        "monika_card": next_card_to_play,
+                        "player_card": self.game.game_log[-2]["played_card"],
+                        "seen_count": 0,
+                        "shown": False
```

Comment:
Is there a point of the `shown` key here?

Ideally if `seen_count > 0`, it is shown. No?

Reply:
Yeah, I can see this being unclear. `seen_count` is basically how many times this reaction had a chance to be selected (there was an in-game situation it was appropriate to say it). `shown` *shows* if the reaction passed's random checks and was actually seen by the player.

Reply:
If that's the case, we should name it more appropriately instead of `shown_count`, maybe put it to be something like `show_chances` or `chances_to_be_shown`